src/node_blob.cc

```diff
@@ -414,15 +416,26 @@ void Blob::StoreDataObject(const v8::FunctionCallbackInfo<v8::Value>& args) {
         std::string(*type, type.length())));
 }
 
-void Blob::RevokeDataObject(const v8::FunctionCallbackInfo<v8::Value>& args) {
+// TODO(@anonrig): Add V8 Fast API to the following function
+void Blob::RevokeObjectURL(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GE(args.Length(), 1);
+  CHECK(args[0]->IsString());
   BlobBindingData* binding_data = Realm::GetBindingData<BlobBindingData>(args);
-
   Environment* env = Environment::GetCurrent(args);
-  CHECK(args[0]->IsString());  // ID key
+  Utf8Value input(env->isolate(), args[0].As<String>());
+  auto out = ada::parse<ada::url_aggregator>(input.ToStringView());
 
-  Utf8Value key(env->isolate(), args[0]);
+  if (!out) {
+    return;
+  }
 
-  binding_data->revoke_data_object(std::string(*key, key.length()));
+  // TODO(@anonrig): Remove this casting when SplitString returns
+  // std::string_view
+  auto path = std::string(out->get_pathname());
+  auto sections = SplitString(path, ':', false);
```

Comment:
If get_pathname returns a std::string_view, then I recommend doing as follows...

![image](https://user-images.githubusercontent.com/391987/234615636-7b4bc64f-e53f-4009-bfa8-ad37ac756f22.png)

In the common case, all you are doing is a search for ':' in a string view and most times the computation will end there.

A search for a single character in a memory region is likely highly optimized by the compiler.